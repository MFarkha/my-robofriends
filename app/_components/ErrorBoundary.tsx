"use client";

import React from "react";
import { ReactNode, Component, ErrorInfo } from "react";
interface IErrorProps {
  children?: ReactNode;
}
interface IErrorState {
  hasError: boolean;
}

class ErrorBoundary extends Component<IErrorProps, IErrorState> {
  constructor(props: IErrorProps) {
    super(props);
  }

  state: IErrorState = {
    hasError: false,
  };

  static getDerivedStateFromError() {
    // Update state so the next render will show the fallback UI.
    return { hasError: true };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    this.setState({ hasError: true });
    console.log("Uncaught", error, errorInfo);
  }

  render() {
    if (this.state.hasError) {
      return (
        <>
          <h1 className="f1">Ooops. Something weird happened.</h1>
        </>
      );
    }
    return this.props.children;
  }
}

export default ErrorBoundary;
