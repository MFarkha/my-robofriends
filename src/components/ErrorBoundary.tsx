import { ReactNode, Component, ErrorInfo } from "react";

interface IErrorProps {
    children?: ReactNode
};

interface IErrorState {
    hasError: boolean
};

class ErrorBoundary extends Component<IErrorProps, IErrorState> {
    state: IErrorState = {
        hasError: false
    }

    componentDidCatch(error: Error, errorInfo: ErrorInfo) {
        this.setState({ hasError: true });
        // console.log("Uncaught", error, errorInfo);
    }

    render() {
        if (this.state.hasError) {
            return <h1>Ooops. Something weird happened.</h1>
        }
        return this.props.children
    }
}

export default ErrorBoundary;