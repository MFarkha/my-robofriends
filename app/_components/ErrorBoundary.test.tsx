import ErrorBoundary from './ErrorBoundary';
import { mount } from 'enzyme';

describe("ErrorBoundary", () => {
    it("should display an ErrorMessage if wrapped component throws", () => {
        const InnerComponent = () => <div>Successful render</div>;
        const wrapper = mount(
        <ErrorBoundary>
            <InnerComponent/>
        </ErrorBoundary>
        );
        const error = new Error("some error");
        wrapper.find(InnerComponent).simulateError(error);
        expect(wrapper.state('hasError')).toBeTruthy();
        expect(wrapper.find(ErrorBoundary)).toMatchSnapshot();
    });
});