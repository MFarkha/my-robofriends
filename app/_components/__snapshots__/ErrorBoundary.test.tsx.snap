// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should display an ErrorMessage if wrapped component throws an error 1`] = `
<div>
  <h1
    class="f1"
  >
    Ooops. Something weird happened.
  </h1>
</div>
`;
